library/Fission/Error.hs

```diff
@@ -25,3 +27,20 @@ fromMaybe err okHandler = maybe (openLeft err) (Right . okHandler)
 
 fromMaybe' :: IsMember err errs => err -> Maybe a -> Either (OpenUnion errs) a
 fromMaybe' err = fromMaybe err identity
+
+withRetryM :: Monad m => Natural -> m (Either err a) -> m (Either err a)
+withRetryM 0 action = action
+withRetryM n action = action >>= \case
+  Left  _err -> withRetryM (n - 1) action
+  Right val  -> return (Right val)
+
+retryForever :: Monad m => (err -> m _ignored) -> m (Either err a) -> m a
+retryForever onErr action = go
+  where
+    go = action >>= \case
+      Right val ->
+        return val
+
+      Left err -> do
+        _ <- onErr err
+        go
```

Comment:
I'm not sure that these are actually used anymore 😅 (from previous commits on this branch)